server/service/service_hosts.go

```diff
@@ -43,7 +43,7 @@ func (svc Service) GetHost(ctx context.Context, id uint) (*fleet.HostDetail, err
 }
 
 func (svc Service) HostByIdentifier(ctx context.Context, identifier string) (*fleet.HostDetail, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionRead); err != nil {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
```

Comment:
Not sure I understand the difference between read and list - instinctively, I'd expect it to be somewhat similar to Get/Select in `sqlx`, i.e. read == get == return a single entity, list == select == return all entities matching some criteria. In that sense, that change is surprising to me, but that may be totally fine.

Reply:
Yeah, the reason I went for it, is because List can be done by any user, but read needs to be done by a user that belongs to a team. So in order to check that, I would have to either get the host (bypassing the authorize) and then authorize that host (which we do below). Or grab all the teams the user belongs to, add it to the host one by one, and see if any are authorized (regardless of whether the host belongs to that team or not).

So enforcing List before getting the host felt like the safest choice.